### ISSUE TITLE
full_width: true error for width="auto"

### ISSUE THREAD

**Issue by arnoutaertgeerts** [mozilla/metrics-graphics#428]
I'm getting an invalid NaN error when using the full_width: true option for an element where the width is set by the "auto" option. How can I fix this?

**Comment by arnoutaertgeerts** [comment]
`d3.select(target).style(dimension)` is returning `"auto"` making `.replace(/px/g, '')` returning `NaN`. Can this function be replaced?

**Comment by hamilton** [comment]
Good catch. I think the function can be set so that when the parent element is "auto" we could do something intelligent.

**Comment by hamilton** [comment]
@arnoutaertgeerts could you perhaps show a small case where this is a problem? I'm having trouble replicating this.

**Comment by AndreMiras** [comment]
I also had the issue. I was using an inline ```<span>``` element rather than using a block ```<div>``` element as target.

**Comment by shendaras** [comment]
I have a terrible hack to fix this issue.  I'm hesitant to include it as a pull request or anything like that as I'm certain there are nicer ways to do it.  I'm a bit of a novice at JavaScript.  This does appear to work for my case, though:

```javascript
function get_pixel_dimension(target, dimension) {
  var node = d3.select(target);
  var sz = node.style(dimension);

  while ((sz == "auto") && node) {
    node = d3.select(node.property('parentNode'));
    if (node) {
      sz = node.style(dimension);
    }
  }

  if ( !node) {
    return Number("NaN");
  }

  return Number(sz.replace(/px/g, ''));
}
```
